packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEdit.tsx

```diff
@@ -98,12 +102,18 @@ const useDraftInboundLines = (itemId: string) => {
     }
   }, [lines, itemId]);
 
+  useEffect(() => {
+    if (!initialLines && draftLines.length > 0) {
+      setInitialLines([...draftLines]);
+    }
+  }, [draftLines]);
```

Comment:
Could be simpler to just use a ref

```
const isDirty = useRef(false)
...
```

in the `updateDraftLine` and `addBatch` functions

```
isDirty.current = true
```

I think?

Reply:
If that does work - maybe a custom hook in common `const {isDirty, markDirty } = useDirtyCheck()` or something?

Reply:
the intention was to check the condition of returning to the original state. if you changed, then reverted, you shouldn't get prompted. Given that I've not been able to remove the confirmation once set, this is a bit pointless though

Reply:
Ah! Of course!

Also by using a state variable, we can react to the 'dirtiness' if ever needed - i.e. if we needed to enable a 'save' button or show a 'not saved' / 'saved' label